cclunit-framework-source/doc/CCLUTMOCKING.md

```diff
@@ -0,0 +1,284 @@
+   
+   ```
+   call cclutAddMockImplementation("testSubroutine", "mockSubroutine")
+   ```
+
+4. Tables can be mocked even if the table does not exist in the domain where the test is run.  The mocked version will be used when executing programs with executeProgramWithMocks.  This can be useful to test tables that do not formally exist yet.
+
+5. The mocked items created through cclutCreateMockTable() and cclutAddMockImplementation will not be applied to child scripts called from the script-under-test.  Some alternatives would be to mock the child script to return the appropriate data or to mock the child script to execute the real script applying the mocked tables and implementations.
+
+6. The mocked items created through cclutCreateMockTable() and cclutAddMockImplementation will not be applied to statements executed through "call parser()" commands.  One alternative would be to separate the parser string generation into separate subroutines and mock the subroutines to return parser strings using the mocked entity names.  Another alternative would be to mock the parser() call to validate the correct information is supplied, then perform the appropriate mock versions of the actions the statement would normally perform.
+
+7. The table mocking APIs are not supported when called from within a reportwriter section.  It might be tempting to use a dummyt query to set up mock data from a record structure, but various mocking calls such as cclutCreateMockTable, cclutRemoveMockTable and cclutAddMockData cannot be executed within the context of a query (because the implementations execute queries). Use a for loop instead.
+ 
+8. Mocking the tdbexecute "reply_to" entity is unsupported under certain conditions, specifically if a call to free the "reply_to" entity is made just prior to calling tdbexecute.  If the scenario is truly necessary for a test, the best alternative is to define and use a subroutine for freeing the "reply_to" entity within the script and use a mock for that subroutine which does not actually perform the freeing of the "reply_to" entity.
+  
+## Example
+Below is an example of some of the APIs available in the CCL Unit Mocking framework along with some simple notes.
+
+Script-under-test:
+
+    drop program 1abc_mo_get_persons:dba go
+    create program 1abc_mo_get_persons:dba    
+    
+    declare newSize = i4 with protect, noconstant(0)
+    
+    select into "nl:"
+    from person p
+    plan p
+	order by p.person_id
+	detail
+		newSize = newSize + 1
+		stat = alterlist(reply->persons, newSize)
+		reply->persons[newSize].person_id = p.person_id
+		reply->persons[newSize].name_last = p.name_last
+		reply->persons[newSize].name_first = p.name_first
+		reply->persons[newSize].birth_dt_tm = p.birth_dt_tm
+	with nocounter
+    
+    end
+    go
+
+Test Code:
+
+    declare mock_table_person = vc with protect, noconstant("")
+    
+    ; Defining a mock person table.  The return value is the name of the mocked table.  
```

Comment:
Perhaps this comment about the return value from cclutDefineMockTable should just be a final statement in the documentation for cclutDefineMockTable. 

Returns a string that is the name of the mock table. This.... 

Note: it should be "mock table" rather than "mocked table"

Reply:
I have an @returns tag in the cclutDefineMockTable documentation.  Are you saying you want this included in the "description" part?